tests/test_netcdf_convert.py

```diff
@@ -90,3 +129,153 @@ def test_unknown_band_selection(in_bands, temp_dir, smap_file, logger):
             in_bands,
             logger
         )
+
+
+def test_non_spatial_variable_fails(temp_dir, logger, nested_file):
+    """Verify a request for a non-spatial variable raises expected exception."""
```

Comment:
This could happen if a user explicitly requests a variable that does not have spatial dimensions to be transformed.